builder/azure/arm/step_publish_to_shared_image_gallery.go

```diff
@@ -36,14 +37,24 @@ func NewStepPublishToSharedImageGallery(client *AzureClient, ui packer.Ui, confi
 	return step
 }
 
-func (s *StepPublishToSharedImageGallery) publishToSig(ctx context.Context, mdiID string, miSigPubRg string, miSIGalleryName string, miSGImageName string, miSGImageVersion string, miSigReplicationRegions []string, location string, tags map[string]*string) (string, error) {
+func (s *StepPublishToSharedImageGallery) publishToSig(ctx context.Context, mdiID string, miSigPubRg string, miSIGalleryName string, miSGImageName string, miSGImageVersion string, miSigReplicationRegions []string, miSGImageVersionEndOfLifeDate string, miSGImageVersionExcludeFromLatest bool, miSigReplicaCount int32, location string, tags map[string]*string) (string, error) {
 
 	replicationRegions := make([]compute.TargetRegion, len(miSigReplicationRegions))
 	for i, v := range miSigReplicationRegions {
 		regionName := v
 		replicationRegions[i] = compute.TargetRegion{Name: &regionName}
 	}
 
+	var endOfLifeDate *date.Time
+	if miSGImageVersionEndOfLifeDate != "" {
+		parseDate, err := date.ParseTime("2006-01-02T15:04:05.99Z", miSGImageVersionEndOfLifeDate)
+		if err != nil {
+			panic(err)
```

Comment:
Our typical error handling format is to return the error so that Packer can properly clean up after itself. Here's an example of how I'd like to see this handled. 

```suggestion
		s.say(fmt.Sprintf("Error parsing date from shared_gallery_image_version_end_of_life_date: %s", err))
		return "", err
```

Reply:
Thanks @SwampDragons. I have made the changes as per your suggestion.